### ISSUE TITLE
need multiple iterations of migration script to be in sync

### ISSUE THREAD

**Issue by mbaroukh** [typeorm/typeorm#6595]
With mysql when creating a new index on an existing entity, the migration:create command generate a ghost indexe that should not exist.

[X] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[X] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[X] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

I created a simple project here : 
https://github.com/mbaroukh/typeorm_synchro
testing need to be done on mysql as there is no problem with sqlite.
You just need to change database configuration here : 
https://github.com/mbaroukh/typeorm_synchro/blob/master/src/test.ts#L17
then execute
```bash
yarn
yarn test mysql
```

It can be run with sqlite or mysql.
With sqlite, no problem.

But here is what happen with mysql :

1/ First iteration give this query to execute : 
```sql
CREATE TABLE `my_entity` (`id` int NOT NULL, `name` varchar(255) NOT NULL, PRIMARY KEY (`id`)) ENGINE=InnoDB
```
The corresponding entity : 
```ts
@Entity()
class MyEntity {
  @PrimaryColumn()
  id!: number;
  @Column()
  name!: string;
}
```
https://github.com/mbaroukh/typeorm_synchro/blob/master/src/step1.ts#L5

At this point, everything is perfect.

2/ Then I just add a unique key on the entity: 

```ts
@Entity()
@Unique("UNIQ_myentity_name", ["name"])
export class MyEntity {
  @PrimaryColumn()
  id!: number;
  @Column()
  name!: string;
}
```
https://github.com/mbaroukh/typeorm_synchro/blob/master/src/step2.ts#L10

The migration generate this : 
```sql
EXECUTING Query 0: ALTER TABLE `my_entity` ADD UNIQUE INDEX `IDX_4b24f3d959c87435c5d89fd102` (`name`)
EXECUTING Query 1: CREATE UNIQUE INDEX `UNIQ_myentity_name` ON `my_entity` (`name`)
```
there is the problem : I don't understand why the first index is created. There should only be mine.

3/ And moreover : At this point, I expect my code to be "in sync" with the database.
But executing the generation again returns this order : 
```sql
DROP INDEX `IDX_4b24f3d959c87435c5d89fd102` ON `my_entity`
```
So it seems that typeorm wasn't aware more than me of why this index exists :)

I understand that we should always check the generated code, and I do. But I report this because I have another problem, more complicated on development mode with ghost indexes when using `synchronize: true` and when trying to create a reproduction case I found this strange behaviour. I think it can be related.

Notice that when generating the second entity from scratch works correctly.
There is only one index created : 
```sql
CREATE TABLE `my_entity` (`id` int NOT NULL, `name` varchar(255) NOT NULL, UNIQUE INDEX `UNIQ_myentity_name` (`name`), PRIMARY KEY (`id`)) ENGINE=InnoDB
```

Thank you

**Comment by imnotjames** [comment]
Duplicate of #3280

**Comment by imnotjames** [comment]
Pretty sure it is, at least.  That was fixed in 0.2.28
